dirty_cat/_feature_augmenter.py

```diff
@@ -37,11 +37,13 @@ class FeatureAugmenter(BaseEstimator, TransformerMixin):
     main_key : str
         The key column name in the main table on which
         the join will be performed.
-    match_score : float, default=0
+    match_distance : float, default=0.0
         Distance score between the closest matches that will be accepted.
         In a [0, 1] interval. 1 means that only a perfect match will be
         accepted, and zero means that the closest match will be accepted,
         no matter how distant.
+        For numerical joins, this defines the maximum euclidian distance
+        between the matches.
```

Comment:
This parameter name is no longer correct. I remember we were unsure about how to name it, and it seems that we got it wrong. 
I guess going back to the 'distance' idea seems right now. My only concern is how to do it, deprecate it, go for a direct change or leave it for strings and create a new for numerical columns?